pkg/sdkserver/sdkserver_test.go

```diff
@@ -112,7 +109,18 @@ func TestSidecarRun(t *testing.T) {
 				assert.NoError(t, err)
 			},
 			expected: expected{
-				state: v1alpha1.GameServerStateAllocated,
+				state:      v1alpha1.GameServerStateAllocated,
+				recordings: []string{string(v1alpha1.GameServerStateAllocated)},
+			},
+		},
+		"reserved": {
+			f: func(sc *SDKServer, ctx context.Context) {
+				_, err := sc.Reserve(ctx, &sdk.Duration{Seconds: 10})
+				assert.NoError(t, err)
+			},
+			expected: expected{
```

Comment:
The expected struct doesn't have a way to check for the expected timer existing. Is there a way to add that check in?

From what I can tell, this is only testing one part of the new functionality: it doesn't test passing in zero and it doesn't verify that the timer properly moves the game server back to ready once the passed in time has elapsed. Will those be covered in tests elsewhere?

What happens if you set seconds to a negative number?

Reply:
it crashes, been there done that.

Reply:
I saw the reset test after I'd written this comment. I think the additional tests below also do a sufficient job making sure the timer exists and fires correctly.